qemu/tests/qemu_guest_agent.py

```diff
@@ -460,6 +461,65 @@ def gagent_check_set_time(self, test, params, env):
                                      " on host")
 
     @error.context_aware
+    def _get_mem_used(self, session, cmd):
+
+        """
+        get memory usage of the process
+
+        :param session: use for sending cmd
+        :param cmd: get details of the process
+        """
+
+        output = session.cmd_output(cmd)
+        logging.info("The process details: %s" % output)
+        try:
+            memory_usage = int(output.split(" ")[-2].replace(",", ""))
+            return memory_usage
+        except Exception:
+            raise exceptions.TestError("Get invalid memory usage by "
+                                       "cmd '%s' (%s)" % (cmd, output))
+
+    @error.context_aware
+    def gagent_check_memory_leak(self, test, params, env):
+
+        """
+        repeat execute "guest-info" command to guest agent, check memory
+        usage of the qemu-ga
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        timeout = float(params.get("login_timeout", 240))
+        test_command = params.get("test_command", "guest-info")
+        memory_usage_cmd = params.get("memory_usage_cmd",
+                                      "tasklist | findstr /I qemu-ga.exe")
+        session = self.vm.wait_for_login(timeout=timeout)
+        error.context("get the memory usage of qemu-ga before run '%s'" %
+                      test_command, logging.info)
+        memory_usage = self._get_mem_used(session, memory_usage_cmd)
+        session.close()
+        repeats = int(params.get("repeats", 1))
+        for i in range(repeats):
+            error.context("execute '%s' %s times" % (test_command, i+1),
+                          logging.info)
+            return_msg = self.gagent.guest_info()
+            logging.info(str(return_msg))
+        self.vm.verify_alive()
+        error.context("get the memory usage of qemu-ga after run '%s'" %
+                      test_command, logging.info)
+        session = self.vm.wait_for_login(timeout=timeout)
+        memory_usage_ = self._get_mem_used(session, memory_usage_cmd)
+        session.close()
+        # less than 20K is acceptable.
+        if memory_usage_ - memory_usage > 20:
+            raise error.TestError("The memory usages are different, "
```

Comment:
Misunderstanding?
I mean shoud use error.TestFail() here, TestError() is usually for env prepare failed.